internal/manager/setup.go

```diff
@@ -42,7 +43,26 @@ func setupLoggers(c *Config) (logrus.FieldLogger, logr.Logger, error) {
 	return deprecatedLogger, logger, nil
 }
 
-func setupControllerOptions(logger logr.Logger, c *Config, scheme *runtime.Scheme) ctrl.Options {
+func setupControllerOptions(logger logr.Logger, c *Config, scheme *runtime.Scheme) (ctrl.Options, error) {
+	// for some features or functionality to work, some additional namespaces (such as system namespaces,
+	// like where the controller manager is running) need to be cached as well.
+	additionalCacheNamespaces := make([]string, 0)
+
+	// gateway APIs controller options setup
+	if c.FeatureGates["Gateway"] {
```

Comment:
This is exactly the kind of branching I was concerned about when I was pushing hard against feature gating :thinking:

I don't have an obvious idea for a fix; let's try to think about it tomorrow.

Reply:
Gateway is a special case because it needs to be aware of the `PublishService` at this stage in the implementation. I don't expect us to have to make any other special adjustments. I also don't think long term as the implementation grows we'll continue to have that limitation, or it least it doesn't need to continue working in the very strict way it does now.

I thought long and hard on this issue before finally implementing this here and in avoidance of building an extra and separate cache just for the Gateway controller (which comes with its own problems) this seemed like the lightest touch for something that is not likely to (and if nothing else I don't think required to) last long term.

Reply:
We're considering this one a blocker for merge after having discussed it in a zoom meeting today.